pkg/hook/controller/kubernetes_bindings_controller.go

```diff
@@ -99,6 +100,57 @@ func (c *kubernetesBindingsController) EnableKubernetesBindings() ([]BindingExec
 	return res, nil
 }
 
+func (c *kubernetesBindingsController) UpdateMonitor(monitorId string, kind, apiVersion string) error {
+	// Find binding for monitorId
+	link, ok := c.BindingMonitorLinks[monitorId]
+	if !ok {
+		return nil
+	}
+
+	bindingName := link.BindingConfig.BindingName
+	// Stop and remove previous monitor instance.
+	err := c.kubeEventsManager.StopMonitor(monitorId)
+	if err != nil {
+		return fmt.Errorf("stop monitor for binding '%s': %v", bindingName, err)
+	}
+
+	// remove snapshots cache for monitor
+	// TODO is it required?
+	delete(c.snapshotsCache, bindingName)
+
+	// Monitor config is a pointer, so is should be updated.
+	if link.BindingConfig.Monitor.Kind != kind {
```

Comment:
I think we should also check the API version here.

Reply:
Done.